### DISCUSSION TITLE
Train Multiple Concepts / Classes

### DISCUSSION THREAD

**Discussion by andykaufseo** [d8ahazard/sd_dreambooth_extension#325] [Q&A] [answered]
Now that we can use [filewords] for instances and classes, does it mean we can train multiple models at the same time with this dreambooth implementation? My captions / filenames contain an exact description of each image.

**Comment by d8ahazard** [comment, marked as answer]
Yes, and even more so with the version I just pushed out.

With the latest, there are now 3x concepts "exposed" to the UI. Each one can use it's own params for classifiers, filewords, max training steps, and sample generation. I need to update the documentation and maybe the hint strings, but I hope it's somewhat straightforward.

**Reply by andykaufseo** [reply]
yea, that's pretty much straightforward, but i have one more quick question (1 month of using dreambooth and could not find the answer to this). when training a new concept, an action (like swimming, jumping or throwing a spear) or a real concept, something more complicated (not a person or an object), what sort of CLASS TOKEN should i use? (for new action or for new concept) Coz the only examples i found are with "dog" or "person / man / woman".

**Reply by d8ahazard** [reply]
That is a good question. I'd probably go with "action" or "activity" or something like that for the class words?

**Reply by axonaxis** [reply]
Can you tell me if it is possible to train 6 or more subjects at once?

**Reply by dugemkakek** [reply]
so the max number of subject can be trained at once is capped at three for now ?

**Reply by d8ahazard** [reply]
> so the max number of subject can be trained at once is capped at three for now ?

Only in the UI. You can create your own JSON file based on the template here:

https://github.com/d8ahazard/sd_dreambooth_extension/blob/main/dreambooth/concepts_list.json

And add as many subjects as your heart desires. You'll need to manually figure out the max steps per subject (or set max steps to 0 and set the number of epochs), but you can literally do as many as you want. 

I just capped it at 3 in the UI because it's a s*itload of variables to pass and keep track of thanks to Gradio being unable to easily support stuff like Object-based data.

**Reply by dugemkakek** [reply]
oh okay, yea was messing on how to add multiple concept not via the UI, nice. thanks for the reply
